vapi/meson.build

```
@@ -431,7 +431,7 @@ gir_bindings += [
 ]
 
 gobject_introspection_dep = dependency('gobject-introspection-1.0', required: false)
```

```suggestion
gobject_introspection_dep = dependency('gobject-introspection-1.0', required: get_option('gir'))
```

^ This would be cleaner, and then change the default to `disabled`.